packages/nuxt/src/app/components/nuxt-link.ts

```diff
@@ -280,8 +279,15 @@ export function defineNuxtLink (options: NuxtLinkOptions) {
 
         // Resolves `to` value if it's a route location object
         // converts `""` to `null` to prevent the attribute from being added as empty (`href=""`)
-        const href = typeof to.value === 'object' ? router.resolve(to.value)?.href ?? null : to.value || null
+        let href = typeof to.value === 'object' ? router.resolve(to.value)?.href ?? null : to.value || null
 
+        // joins with `baseURL` if it's an relative link
+        if (import.meta.client && href && href.startsWith('/') && !href.startsWith('//')) {
```

Comment:
Why check only on the client side? This could lead to inconsistent results between the server side and the client side.

```suggestion
-        if (import.meta.client && href && href.startsWith('/') && !href.startsWith('//')) {
+        if (href && href.startsWith('/') && !href.startsWith('//')) {
```

Reply:
Could you please review it once more? I haven't personally encountered the situation where `useRuntimeConfig` is unable to retrieve the `baseURL` on the server side in my environment.

Reply:
Have you tried this command?

```
pnpm vitest nuxt-link
```

In my environment, the test does not pass without detecting `import.meta.client`:

```
 FAIL  packages/nuxt/test/nuxt-link.test.ts > nuxt-link:propsOrAttributes > `isExternal` is `true` > href > resolves route location object
Error: [nuxt] instance unavailable
 ❯ useNuxtApp packages/nuxt/src/app/nuxt.ts:395:13
    409| }
    410|
    411| export function defineAppConfig<C extends AppConfigInput> (config: C): C {
       |                                             ^
    412|   return config
    413| }
 ❯ Module.useRuntimeConfig packages/nuxt/src/app/nuxt.ts:404:10
 ❯ packages/nuxt/src/app/components/nuxt-link.ts:286:27
 ❯ nuxtLink packages/nuxt/test/nuxt-link.test.ts:46:96
 ❯ packages/nuxt/test/nuxt-link.test.ts:115:16

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/3]⎯

 Test Files  1 failed (1)
      Tests  3 failed | 25 passed (28)
   Start at  14:31:52
   Duration  2.81s (transform 179ms, setup 2.22s, collect 167ms, tests 13ms, environment 0ms, prepare 155ms)
```